timemachine/md/minimizer.py

```diff
@@ -451,8 +453,35 @@ def val_and_grad_fn(coords):
     return val_and_grad_fn
 
 
+def wrap_val_and_grad_with_positional_restraint(
+    val_and_grad_fn: Callable[[NDArray], Tuple[float, NDArray]],
+    x0: NDArray,
+    box0: NDArray,
+    free_idxs: NDArray,
+    k: float,
+) -> Callable[[NDArray], Tuple[float, NDArray]]:
+    restraint_val_and_grad = jax.value_and_grad(harmonic_positional_restraint, argnums=1)
+
+    starting_free = np.array(x0[free_idxs])
+
+    def wrapped_minimization(x):
```

Comment:
naming: `wrapped_val_and_grad` or similar

Reply:
Changed in https://github.com/proteneer/timemachine/pull/1393/commits/8ec202fad2f1345b47d300d1a5f4104ea2e23673